frontend/src/js/components/modals/TemplateEditor.jsx

```diff
@@ -22,6 +27,9 @@ var TemplateEditor = React.createClass({
                             </h4>
                         </div>
                         <TemplatePreview template={template} />
+                        <TemplateLabelsEditor template={template} updateTemplate={this.onUpdate}/>
+                        <button type="button" aria-label="Save">Save</button>
+                        <button type="button" aria-label="Cancel">Cancel</button>
                     </div>
```

Comment:
save needs to save state `this.props.template = this.state.workingTemplate`
cancel needs to revert and close 'this.state.workingTemplate = his.props.template` && close